docs/4.4/api-reference.md

```diff
@@ -7,61 +7,479 @@ description: The detailed guide to Teleport API
+}
+```
+
+### Retrieve Role
+
+This is the equivalent of `tctl get role/admin`.
+
+```go
+role, err := client.GetRole("admin")
+if err != nil {
+  return err
+}
+```
+
+### Create Role
+
+You can use the UpsertRole RPC to programmatically create a new role. This is the equivalent of `tctl create -f auditor-role.yaml`, where the `-f` flag signals to overwrite the auditor role if it exists already.
+
+For example, suppose you wanted to create a role for an auditor that could view all sessions, but could not access any servers.
+
+Using tctl, you would first create a role that allows reading and listing of the session resource like below. In addition, the user is explicitly denied access to all nodes in the deny block.
+
+```
+$ cat << EOF > /tmp/auditor-role.yaml
 kind: role
+version: v3
 metadata:
-  name: access-plugin
+  name: auditor
 spec:
+  options:
+    max_session_ttl: 8h
   allow:
     rules:
-      - resources: ['access_request']
-        verbs: ['list','read','update']
-    # teleport currently refuses to issue certs for a user with 0 logins,
-    # this restriction may be lifted in future versions.
-    logins: ['access-plugin']
-version: v3
+    - resources: [session]
+      verbs: [list, read]
+  deny: {}
+    node_labels: '*': '*'
 EOF
-# ...
-$ tctl create rscs.yaml
-# ...
-$ tctl auth sign --format=tls --user=access-plugin --out=auth
-# ...
+$ tctl create -f /tmp/auditor-role.yaml
+```
+
+To do something similar with the API:
+
+```go
+role, err := services.NewRole("auditor", services.RoleSpecV3{
+  Options: services.RoleOptions{
+    MaxSessionTTL: services.Duration(time.Hour),
+  },
+  Allow: services.RoleConditions{
+    Logins: []string{"auditor"},
+    Rules: []services.Rule{
+      services.NewRule(services.KindSession, services.RO()),
+    },
+  },
+  Deny: services.RoleConditions{
+    NodeLabels: services.Labels{"*": []string{"*"}},
+  },
+})
+if err != nil {
+  return err
+}
+if err = client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+### Update Role
+
+The UpsertRole RPC can also be used to update an existing role.
+
+You can update practically any field within the role struct using the role's setter functions.
+
+```go
+// retrieve role
+role, err := client.GetRole("auditor")
+if err != nil {
+  return err
+}
+
+// update the auditor role's ttl to one day
+role.SetOptions(services.RoleOptions{
+  MaxSessionTTL: services.Duration(time.Hour * 24),
+})
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+### Delete Role
+
+This is the equivalent of `tctl rm auditor-role.yaml`.
+
+```go
+if err := client.DeleteRole(ctx, "auditor"); err != nil {
+  return err
+}
+```
+
+## Tokens
+
+[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add leaf clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+
+### Retrieve token
+
+```go
+token, err := client.GetToken(tokenString)
+if err != nil {
+  return err
+}
+```
+
+### Create token
+
```

Comment:
Can you expand that create token is similar to ` tctl tokens add` and it accepts 'node, trusted_cluser and proxy`. It can also accept a --value. You might want to add that here.